fix: Disable Invite Members button if org requires SSO

RyanSkonnord opened the pull request:
Requires https://github.com/getsentry/sentry/pull/73593

Resolves https://github.com/getsentry/sentry/issues/56889

RyanSkonnord committed:
fix: Disable Invite Members button if org requires SSO

RyanSkonnord commented:
Up for preliminary review. Still needs some new unit tests, and probably to be split into front-end and back-end PRs.

RyanSkonnord committed:
Merge branch 'master' into disable-invite-members-per-sso

cathteng commented on static/app/types/organization.tsx:
```diff
@@ -70,6 +70,7 @@ export interface Organization extends OrganizationSummary {
     projectLimit: number | null;
   };
   relayPiiConfig: string | null;
+  requiresSso: boolean;
```
if we merge BE and FE at the same time, this is not guaranteed to be sent from the backend when the frontend is merged, especially since backend deploys take longer. it might error unless we also allow `null` here

RyanSkonnord replied:
Would it be more sensible to do `requiresSso?`. I see no problem treating `null` and `false` interchangeably.

cathteng replied:
i think that works too

evanpurkhiser replied:
Let's just deploy the backend first.

RyanSkonnord committed:
Add front-end unit test

RyanSkonnord committed:
Add back-end unit test

RyanSkonnord committed:
Merge branch 'master' into disable-invite-members-per-sso

RyanSkonnord committed:
Move back-end changes to another PR

RyanSkonnord commented:
Back end changes split into https://github.com/getsentry/sentry/pull/73593. Now ready for review.

RyanSkonnord committed:
Add requiresSso to OrganizationFixture

RyanSkonnord commented:
Hmm, I'm getting a type error due to https://github.com/getsentry/getsentry/blob/master/static/getsentry/gsApp/components/inviteMembersButtonCustomization.tsx#L4

So there's a chicken-and-egg problem in syncing the type change between Sentry and Getsentry. AFAICT a simple fix would be not to change the name of `disabled` and to make `disabledBySso` optional. But I prefer not to have `disabled` be ambiguous like that. If there's a good way to sync up the changes otherwise, let me know.

evanpurkhiser commented on static/app/types/hooks.tsx:
```diff
@@ -631,10 +631,8 @@ type IntegrationsFeatureGatesHook = () => {
-     * Whether the Invite Members button is active or not
-     */
-    disabled: boolean;
+    disabledByFlag: boolean;
+    disabledBySso: boolean;
```
Can we add comments? What is "disabled by flag"

evanpurkhiser replied:
```suggestion
    disabledByFlag: boolean;
    disabledBySso: boolean;
```

evanpurkhiser commented on static/app/types/hooks.tsx:
```diff
@@ -631,10 +631,8 @@ type IntegrationsFeatureGatesHook = () => {
-     * Whether the Invite Members button is active or not
-     */
-    disabled: boolean;
+    disabledByFlag: boolean;
+    disabledBySso: boolean;
```
I think you'll need to keep the `disabled` around for BC otherwise getsentry is going to fail no?

RyanSkonnord replied:
Yeah, that's essentially what I was asking about [here](https://github.com/getsentry/sentry/pull/73521#issuecomment-2200739255). I'll revert the name change.

evanpurkhiser commented on static/app/views/settings/organizationMembers/organizationMembersWrapper.tsx:
```diff
@@ -172,6 +175,19 @@ function renderInviteMembersButton({
+      body={
+        <Fragment>
+          {t(
+            `Your organization must use its single sign-on provider to invite
+            new members.`
```
Is the phrasing here right? Are they "inviting" members via SSO, or are they "registering" members?

```suggestion
            `Your organization must use its single sign-on provider to register new members.`
```

evanpurkhiser commented on static/app/views/settings/organizationMembers/organizationMembersWrapper.tsx:
```diff
@@ -172,6 +175,19 @@ function renderInviteMembersButton({
       {action}
     </Hovercard>
+  ) : disabledBySso ? (
+    <Hovercard
```
Should this be a hovercard? Feels like this should just be a tooltip?

RyanSkonnord replied:
Sure. I was just imitating the other "disabled" message.

Any further, subjective tweaks to phrasing or presentation are welcome. I was just winging it on the message to the user.

RyanSkonnord committed:
Change message phrasing

Co-authored-by: Evan Purkhiser <evanpurkhiser@gmail.com>

RyanSkonnord committed:
Merge branch 'master' into disable-invite-members-per-sso

RyanSkonnord committed:
Restore `disabled` param with original meaning

Rename `disabledBySso` to `isSsoRequired`, which now controls only the
displayed message.

Lines starting # with '#' will be ignored, and an empty message aborts
the commit. # # On branch disable-invite-members-per-sso # Changes to be
committed: # modified: static/app/types/hooks.tsx # modified:
static/app/views/settings/organizationMembers/organizationMembersWrapper.tsx

RyanSkonnord committed:
Change SSO message to tooltip

evanpurkhiser commented on static/app/views/settings/organizationMembers/organizationMembersWrapper.tsx:
```diff
@@ -161,17 +164,27 @@ function renderInviteMembersButton({
-      {action}
-    </Hovercard>
+    isSsoRequired ? (
+      <Tooltip
+        title={t(
+          `Your organization must use its single sign-on provider to register new members.`
+        )}
+      >
```
I think you can add `skipWrappper` here
```suggestion
      <Tooltip
        skipWrapper
        title={t(
          `Your organization must use its single sign-on provider to register new members.`
        )}
      >
```

RyanSkonnord committed:
Merge branch 'master' into disable-invite-members-per-sso

RyanSkonnord committed:
Add skipWrapper to tooltip

Co-authored-by: Evan Purkhiser <evanpurkhiser@gmail.com>

RyanSkonnord committed:
Merge branch 'master' into disable-invite-members-per-sso
